hibernate-orm-modules/src/test/java/org/hibernate/wildfly/integrationtest/JavassistHibernateModulesOnWildflyTest.java

```diff
@@ -0,0 +1,107 @@
+ */
+package org.hibernate.wildfly.integrationtest;
+
+import java.util.Properties;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+
+import org.hibernate.Session;
+import org.hibernate.bytecode.internal.javassist.BytecodeProviderImpl;
+import org.hibernate.bytecode.spi.BasicProxyFactory;
+import org.hibernate.bytecode.spi.BytecodeProvider;
+import org.hibernate.bytecode.spi.ProxyFactoryFactory;
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.cfg.Environment;
+import org.hibernate.wildfly.model.Kryptonite;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.Asset;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.shrinkwrap.descriptor.api.Descriptors;
+import org.jboss.shrinkwrap.descriptor.api.persistence21.PersistenceDescriptor;
+import org.jboss.shrinkwrap.descriptor.api.persistence21.PersistenceUnitTransactionType;
+
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.hamcrest.core.IsNull.notNullValue;
+import static org.junit.Assert.assertThat;
+
+/**
+ * The purpose of this test is to check that it's still possible to use Javassist as byte code provider with WildFly.
+ */
+@RunWith(Arquillian.class)
+public class JavassistHibernateModulesOnWildflyTest {
+
+	private static final String ORM_VERSION = Session.class.getPackage().getImplementationVersion();
+	private static final String ORM_MINOR_VERSION = ORM_VERSION.substring( 0, ORM_VERSION.indexOf( ".", ORM_VERSION.indexOf( "." ) + 1) );
+
+	@Deployment
+	public static WebArchive createDeployment() {
+		return ShrinkWrap.create( WebArchive.class )
+				.addClass( Kryptonite.class )
+				.addAsWebInfResource( EmptyAsset.INSTANCE, "beans.xml" )
+				.addAsResource( persistenceXml(), "META-INF/persistence.xml" );
+	}
+
+	private static Asset persistenceXml() {
+		PersistenceDescriptor persistenceXml = Descriptors.create( PersistenceDescriptor.class )
+					.version( "2.1" )
+					.createPersistenceUnit()
+						.name( "primary" )
+						.transactionType( PersistenceUnitTransactionType._JTA )
+						.jtaDataSource( "java:jboss/datasources/ExampleDS" )
+						.getOrCreateProperties()
+							// We want to use the ORM from this build instead of the one coming with WildFly
+							.createProperty().name( "jboss.as.jpa.providerModule" ).value( "org.hibernate:" + ORM_MINOR_VERSION ).up()
+							.createProperty().name( "hibernate.hbm2ddl.auto" ).value( "create-drop" ).up()
+							.createProperty().name( "hibernate.allow_update_outside_transaction" ).value( "true" ).up()
+							.createProperty().name( AvailableSettings.BYTECODE_PROVIDER ).name( Environment.BYTECODE_PROVIDER_NAME_JAVASSIST ).up()
```

Comment:
Are you sure you can set `BYTECODE_PROVIDER` this way? This property is read only from system properties when the Hibernate ORM classes are first initialized.

You seem to have noticed below, as the test is setting it in a different way.

I'd suggest to use the Arquillian capability to set JVM properties.

Reply:
How would you set it on Arquillian?

Reply:
Forget about it, silly question

Reply:
https://github.com/hibernate/hibernate-orm/blob/9b08e6bc290d5e35dfa2344dead19f1918ce11c9/hibernate-orm-modules/src/test/resources/arquillian.xml#L28

Reply:
I'm not going to update the arquillian.xml because I don't want the change to affect the other tests as well. I've also added a coment in the test.